PowerShellTools/Intellisense/AutoCompletionController.cs

```diff
@@ -52,34 +52,59 @@ internal sealed class AutoCompletionController : IOleCommandTarget
             _editorOperations = editorOperations;
             _undoHistory = undoHistory;
             _serviceProvider = serviceProvider;
+            _autoCompleteCount = 0;
         }
 
         public IOleCommandTarget NextCommandHandler { get; set; }
 
+        private bool IsLastCmdAutoComplete
+        {
+            get
+            {
+                return _autoCompleteCount > 0;
+            }
+            set
+            {
+                if (value)
+                {
+                    _autoCompleteCount++;
+                }
+                else
+                {
+                    _autoCompleteCount = 0;
+                }
+            }
+        }
+
         #region IOleCommandTarget implementation
 
         public int Exec(ref Guid pguidCmdGroup, uint nCmdID, uint nCmdexecopt, IntPtr pvaIn, IntPtr pvaOut)
         {
+            var command = (VSConstants.VSStd2KCmdID)nCmdID;
+
             if (VsShellUtilities.IsInAutomationFunction(_serviceProvider) ||
                 pguidCmdGroup != VSConstants.VSStd2K ||

```

Comment:
Can we pull some of these Gnarly ifs into their own private function for documentation/readability's sake?